baseband/vlbi_base/base.py

```diff
@@ -109,45 +89,70 @@ class initializer.
         """
         return self._subset
 
-    def _get_sample_shape(self, subset_wrapints):
-        # Extract sample_shape by creating a dummy sample and indexing it
-        # with subset_wrapints.
-        dummy_sample = np.empty(self._unsliced_shape)
+    def _get_sample_shape(self):
+        # Extract semple_shape by creating a possibly squeezed dummy sample,
+        # and then indexing it with subset.
+        sample_shape = self._unsliced_shape
+        fields = getattr(sample_shape, '_fields', None)
+        if self.squeeze:
+            squeezed_shape = tuple(dim for dim in sample_shape if dim > 1)
+            if fields is None:
+                sample_shape = squeezed_shape
+            else:
+                fields = [field for field, dim in zip(fields, sample_shape)
+                          if dim > 1]
+                shape_cls = namedtuple('SampleShape', ','.join(fields))
+                sample_shape = shape_cls(*squeezed_shape)
+
+        if not self.subset:
+            return sample_shape
+
+        # Create a dummy sample that has the sample number as its value.
+        dummy_data = np.arange(13.)
+        dummy_sample = np.rollaxis(
```

Comment:
Unimportant, but apparently `np.rollaxis` is deprecated, and we eventually will want to shift to `np.moveaxis(x, -1, 0)`.

Reply:
Yes, after our minimum version is 1.11